### PR TITLE
Delegations have a field named `name`, not `role`

### PR DESCRIPTION
**Pull request by erickt**
For reference, see the [spec].

[spec]: https://theupdateframework.github.io/specification/latest/#delegations

### CODE REVIEW

File: tuf/src/interchange/cjson/shims.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use chrono::offset::Utc;
 use chrono::prelude::*;
 use serde_derive::{Deserialize, Serialize};
 use std::collections::{BTreeMap, HashSet};
+use std::convert::{TryFrom, TryInto};
```
Comment (JayXon) [inline tuf/src/interchange/cjson/shims.rs]: is `TryInto` needed here? it's included in edition 2021 prelude?

File: tuf/src/interchange/cjson/shims.rs [hunk 2]
```diff
@@ -368,82 +369,109 @@ pub struct PublicKeyValue {
...
+impl From<&metadata::Delegations> for Delegations {
+    fn from(delegations: &metadata::Delegations) -> Delegations {
+        let mut roles = delegations
+            .roles()
+            .iter()
+            .map(Delegation::from)
+            .collect::<Vec<Delegation>>();
+
+        // We want our roles in a consistent order.
+        roles.sort_by(|lhs, rhs| lhs.name.cmp(&rhs.name));
```
Comment (JayXon) [inline tuf/src/interchange/cjson/shims.rs:449]: nit: use `sort_by_key`?

**Review by JayXon — APPROVED** [review]

File: tuf/src/interchange/cjson/shims.rs (hunk 1 above)
Comment (erickt) [reply tuf/src/interchange/cjson/shims.rs]: done

File: tuf/src/interchange/cjson/shims.rs (hunk 2 above)
Comment (erickt) [reply tuf/src/interchange/cjson/shims.rs:449]: I'd love to be able to use `roles.sort_by_key(|r| &r.name)`, but unfortunately that gives this error:

```
449 |         roles.sort_by_key(|role| &role.name);
    |                            ----- ^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
    |                            |   |
    |                            |   return type of closure is &'2 MetadataPath
    |                            has type `&'1 shims::Delegation`
```

Comment (JayXon) [reply tuf/src/interchange/cjson/shims.rs:449]: oh I remember this https://github.com/rust-lang/rust/issues/34162

**Pull request merged**